### PR TITLE
Add path() step

### PR DESCRIPTION
**Pull request by gabi-giladov**
Added support for the 'path' step in terms of query execution and result deserialization.
Added deserialization tests.

### CODE REVIEW

File: src/ExRam.Gremlinq.Core/Queries/Interfaces/IGremlinQuery.cs [hunk 1]
```diff
@@ -123,6 +123,8 @@ public interface IGremlinQueryBaseRec<TSelf> : IGremlinQueryBase
         TSelf Order(Func<IOrderBuilder<TSelf>, IOrderBuilderWithBy<TSelf>> projection);
         TSelf OrderLocal(Func<IOrderBuilder<TSelf>, IOrderBuilderWithBy<TSelf>> projection);
 
+        TSelf Path();
```
Comment (danielcweber) [inline src/ExRam.Gremlinq.Core/Queries/Interfaces/IGremlinQuery.cs:126]: But is it? If I have an `IVertexGremlinQuery<Element>` and call Path on it, the resulting query doesn't have vertex semantics, nor is it typed on `Element` (i.e. query results with a Path step are unlikely able to be deserialized to Element). It's rather something like `IValueGremlinQuery<SomePathClass>` for a reasonably choice of SomePathClass. So it must be decided what this SomePathClass class must look like.

**Review by danielcweber — COMMENTED** [review]
Thanks for the contribution. I have started a review.

File: src/ExRam.Gremlinq.Core/Queries/Interfaces/IGremlinQuery.cs (hunk 1 above)
Comment (gabi-giladov) [reply src/ExRam.Gremlinq.Core/Queries/Interfaces/IGremlinQuery.cs]: Thanks. Fixed and tested

**Comment by danielcweber** [comment]
What's unfortunately completely missing is any test for deserialization. What does raw GraphSON with a path look like? Given some path-GraphSON, does the deserialized Path match the expectation? There's no tests.

**Comment by gabi-giladov** [comment]
> What's unfortunately completely missing is any test for deserialization. What does raw GraphSON with a path look like? Given some path-GraphSON, does the deserialized Path match the expectation? There's no tests.

I tested successfully through the Samples project, using the default deserializer since the Path type is well defined. Can you please give an example of such tests?

**Comment by gabi-giladov** [comment]
> > What's unfortunately completely missing is any test for deserialization. What does raw GraphSON with a path look like? Given some path-GraphSON, does the deserialized Path match the expectation? There's no tests.
> 
> I tested successfully through the Samples project, using the default deserializer since the Path type is well defined. Can you please give an example of such tests?

Added a new test into GraphsonSupportTests.

File: src/ExRam.Gremlinq.Core/Elements/Path.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿using System.Collections.Generic;
+
+namespace ExRam.Gremlinq.Core.GraphElements
+{
+    public sealed class Path
+    {
+        public string[][]? Labels { get; set; }
+        public IElement[]? Objects { get; set; }
```
Comment (danielcweber) [inline src/ExRam.Gremlinq.Core/Elements/Path.cs:8]: I'm afraid it's unfortuately more complicated. Consider the query g.V().out().map(count()).path(). Not all items in a path are necessarily IElements. Furthermore. implementing IElement is not a requirement for entities, so you can't do better than object I guess.

File: test/ExRam.Gremlinq.Core.Tests/Json/Paths_of_Languages.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
+﻿[
```
Comment (danielcweber) [inline test/ExRam.Gremlinq.Core.Tests/Json/Paths_of_Languages.json:1]: Could you add a test where the GraphSON is version >= 2? There's an example here: http://tinkerpop.apache.org/docs/current/dev/io/#_path_2

**Comment by danielcweber** [comment]
Please rebase on the latest main branch and have a look at the failing tests, if still any.

**Comment by danielcweber** [comment]
I'll have the single failing test for a missing GremlinServerIntegrationTest.*.verified file covered (it's because you added a new test for serialization but it didn't produce output for an integration test on Gremlin Server. If you feel brave you can tackle it with a locally running instance of Gremlin Server though and be good ;-)).

File: src/ExRam.Gremlinq.Core/Elements/Path.cs (hunk 2 above)
Comment (gabi-giladov) [reply src/ExRam.Gremlinq.Core/Elements/Path.cs]: You are right. I missed that.
Changed to 'object'.

File: test/ExRam.Gremlinq.Core.Tests/Json/Paths_of_Languages.json (hunk 3 above)
Comment (gabi-giladov) [reply test/ExRam.Gremlinq.Core.Tests/Json/Paths_of_Languages.json]: Added Graphson2 and Graphson3 tests

**Comment by gabi-giladov** [comment]
Thank you for your time and patience! Appreciate it a lot

**Comment by danielcweber** [comment]
Awesome contribution. Thanks a lot!

**Comment by danielcweber** [comment]
Which version of Gremlin Server did you use for the integration test? For 3.4.9, I get a complete different Verify-output. That's why the build on main fails. I need to look into this.

**Comment by gabi-giladov** [comment]
> Which version of Gremlin Server did you use for the integration test? For 3.4.9, I get a complete different Verify-output. That's why the build on main fails. I need to look into this.

We are working on Cosmos DB Gremlin API.

**Comment by danielcweber** [comment]
Got it. The GremlinServerIntegrationTests are meant to be run on GremlinServer (https://www.apache.org/dyn/closer.lua/tinkerpop/3.4.9/apache-tinkerpop-gremlin-server-3.4.9-bin.zip), but I'll have that covered.

**Comment by danielcweber** [comment]
I'd like to have integration tests for CosmosDB as well, but the emulator unfortunately won't run on Linux and hence not within a  Github Actions build.

**Comment by danielcweber** [comment]
@gabi-giladov: There's a preview of version 8.1: https://www.nuget.org/packages/ExRam.Gremlinq.Core/8.1.0-preview-0017

**Pull request merged**